script.js

```diff
@@ -1,4 +1,235 @@
+// === REQUIREMENTS //
+// there are 2 players and players take turns
+// when a player clicks submit, the game rolls 2 dice and shows the dice roll, for example 3 and 6
+// the player picks the order of the dice they want. For example, if they wanted the number 63, they would specify that the 2nd dice goes first. next player would complete all the steps
+// after both players have rolled and chosen dice order, the player with the higher combined number wins
+
+// == problem breakdown and planning === //
+// ver. 1 rolls 2 dice and turns the output for 1 payer. then player chooses the dice order and get the correct return output. final chosen value will reflect in the outpux box
+// ver. 2 refactor the code to work with 2 players
+//      - global variables fort currentPlayer; and an array to store the score of all players allPlayerScore
+//      - refactor outputMessages to interact with each player, 1 and 2
+//      - write logic for player 1 to go first then player 2, and finally point towards comparing score
+//ver 3. implement comparing dice scores and declare winner
+// ver 4. reset the game so that the players can play continually without refreshing the browser page
+
+//declare global variable; all caps to remind that these will never change
+//GLOBAL VARIABLES
+var GAME_STATE_DICE_ROLL = "GAME_STATE_DICE_ROLL";
+var GAME_STATE_CHOOSE_DICE_ORDER = "GAME_STATE_CHOOSE_DICE_ORDER";
+var GAME_STATE_COMPARE_SCORES = "GAME_STATE_COMPARE_SCORES";
+var GAME_STATE_LEADERBOARD = "GAME_STATE_LEADERBOARD";
+//first step of the game
+var gameState = GAME_STATE_DICE_ROLL;
+
+var currentPlayerRolls = [];
+var currentPlayer = 1;
+var allPlayersScore = [];
+var overallPlayersScore = [];
+var player1Scores = [];
+var player2Scores = [];
+var runningPlayerScore1 = 0;
+var runningPlayerScore2 = 0;
+var i = 0;
+var j = 1;
+var k = 0;
+var l = 1;
+var roundCounter = 1;
+
+// helper function = rollDice
+var rollDice = function () {
+  console.log(`Control flow: start of rollDice()`);
```

Comment:
generally for a "completed code" we don't add console logs anymore. It's always best practice to remove these on submission